src/cal/atmcorr_specmath.py

```diff
@@ -67,21 +70,41 @@ def avgs_finder(atmotxt_dir):
     # values
     averages = []
 
-    # Opens the .txt document 
-    with open(atmotxt_dir, 'r') as atmo_txt:
-        # Stores each line as its own list inside a larger list
-        # Splitting is done due to the formatting in the file
-        text = [line.strip().replace("\n", "").split(": ")
-                for line in atmo_txt]
-    # Closes the file
-    atmo_txt.close()
-
-    # Sets a counter variable
+    if not missing_txt:
+        # Opens the .txt document 
+        with open(atmotxt_dir, 'r') as atmo_txt:
+            # Stores each line as its own list inside a larger list
+            # Splitting is done due to the formatting in the file
+            text = [line.strip().replace("\n", "").split(": ")
+                    for line in atmo_txt]
+        # Closes the file
+        atmo_txt.close()
```

Comment:
When you do `with` you do not have to close the file at the end. It gets closed automatically